arch/RISCV/RISCVMapping.c

```diff
@@ -0,0 +1,1918 @@
+
+	{RISCV_REG_X0, "zero"},
+	{RISCV_REG_X1, "ra"},
+	{RISCV_REG_X2, "sp"},
+	{RISCV_REG_X3, "gp"},
+	{RISCV_REG_X4, "tp"},
+	{RISCV_REG_X5, "t0"},
+	{RISCV_REG_X6, "t1"},
+	{RISCV_REG_X7, "t2"},
+	{RISCV_REG_X8, "s0"},
+	{RISCV_REG_X9, "s1"},
+	{RISCV_REG_X10, "a0"},
+	{RISCV_REG_X11, "a1"},
+	{RISCV_REG_X12, "a2"},
+	{RISCV_REG_X13, "a3"},
+	{RISCV_REG_X14, "a4"},
+	{RISCV_REG_X15, "a5"},
+	{RISCV_REG_X16, "a6"},
+	{RISCV_REG_X17, "a7"},
+	{RISCV_REG_X18, "s2"},
+	{RISCV_REG_X19, "s3"},
+	{RISCV_REG_X20, "s4"},
+	{RISCV_REG_X21, "s5"},
+	{RISCV_REG_X22, "s6"},
+	{RISCV_REG_X23, "s7"},
+	{RISCV_REG_X24, "s8"},
+	{RISCV_REG_X25, "s9"},
+	{RISCV_REG_X26, "s10"},
+	{RISCV_REG_X27, "s11"},
+	{RISCV_REG_X28, "t3"},
+	{RISCV_REG_X29, "t4"},
+	{RISCV_REG_X30, "t5"},
+	{RISCV_REG_X31, "t6"},
+
+	{RISCV_REG_F0_32, "ft0"},
+	{RISCV_REG_F1_32, "ft1"},
+	{RISCV_REG_F2_32, "ft2"},
+	{RISCV_REG_F3_32, "ft3"},
+	{RISCV_REG_F4_32, "ft4"},
+	{RISCV_REG_F5_32, "ft5"},
+	{RISCV_REG_F6_32, "ft6"},
+	{RISCV_REG_F7_32, "ft7"},
+	{RISCV_REG_F8_32, "fs0"},
+	{RISCV_REG_F9_32, "fs1"},
+	{RISCV_REG_F10_32, "fa0"},
+	{RISCV_REG_F11_32, "fa1"},
+	{RISCV_REG_F12_32, "fa2"},
+	{RISCV_REG_F13_32, "fa3"},
+	{RISCV_REG_F14_32, "fa4"},
+	{RISCV_REG_F15_32, "fa5"},
+	{RISCV_REG_F16_32, "fa6"},
+	{RISCV_REG_F17_32, "fa7"},
+	{RISCV_REG_F18_32, "fs2"},
+	{RISCV_REG_F19_32, "fs3"},
+	{RISCV_REG_F20_32, "fs4"},
+	{RISCV_REG_F21_32, "fs5"},
+	{RISCV_REG_F22_32, "fs6"},
+	{RISCV_REG_F23_32, "fs7"},
+	{RISCV_REG_F24_32, "fs8"},
+	{RISCV_REG_F25_32, "fs9"},
+	{RISCV_REG_F26_32, "fs10"},
+	{RISCV_REG_F27_32, "fs11"},
+	{RISCV_REG_F28_32, "ft8"},
+	{RISCV_REG_F29_32, "ft9"},
+	{RISCV_REG_F30_32, "ft10"},
+	{RISCV_REG_F31_32, "ft11"},
+
+	{RISCV_REG_F0_64, "ft0"},
+	{RISCV_REG_F1_64, "ft1"},
+	{RISCV_REG_F2_64, "ft2"},
+	{RISCV_REG_F3_64, "ft3"},
+	{RISCV_REG_F4_64, "ft4"},
+	{RISCV_REG_F5_64, "ft5"},
+	{RISCV_REG_F6_64, "ft6"},
+	{RISCV_REG_F7_64, "ft7"},
+	{RISCV_REG_F8_64, "fs0"},
+	{RISCV_REG_F9_64, "fs1"},
+	{RISCV_REG_F10_64, "fa0"},
+	{RISCV_REG_F11_64, "fa1"},
+	{RISCV_REG_F12_64, "fa2"},
+	{RISCV_REG_F13_64, "fa3"},
+	{RISCV_REG_F14_64, "fa4"},
+	{RISCV_REG_F15_64, "fa5"},
+	{RISCV_REG_F16_64, "fa6"},
+	{RISCV_REG_F17_64, "fa7"},
+	{RISCV_REG_F18_64, "fs2"},
+	{RISCV_REG_F19_64, "fs3"},
+	{RISCV_REG_F20_64, "fs4"},
+	{RISCV_REG_F21_64, "fs5"},
+	{RISCV_REG_F22_64, "fs6"},
+	{RISCV_REG_F23_64, "fs7"},
+	{RISCV_REG_F24_64, "fs8"},
+	{RISCV_REG_F25_64, "fs9"},
+	{RISCV_REG_F26_64, "fs10"},
+	{RISCV_REG_F27_64, "fs11"},
+	{RISCV_REG_F28_64, "ft8"},
+	{RISCV_REG_F29_64, "ft9"},
+	{RISCV_REG_F30_64, "ft10"},
+	{RISCV_REG_F31_64, "ft11"},
+};
+#endif
+
+const char *RISCV_reg_name(csh handle, unsigned int reg)
+{
+#ifndef CAPSTONE_DIET
+	if (reg >= RISCV_REG_ENDING)
+		return NULL;
+
+	return reg_name_maps[reg].name;
+#else
+	return NULL;
+#endif
+}
+
+static const insn_map insns[] = {
+	// dummy item
+	{
+	 0, 0,
+#ifndef CAPSTONE_DIET
+	 {0}, {0}, {0}, 0, 0
+#endif
+	 },
+
+	{
```

Comment:
please put everything (in this array) after this line to a new file RISCVMappingInsn.inc, then `#include "RISCVMappingInsn.inc"`  here, like other archs.